/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.carlistingservice;

import "google/protobuf/empty.proto";
import public "models/car_listing.proto";

service CarListingService {
  rpc CarlistingCreate (CarlistingCreateRequest) returns (CarListing);

  rpc CarlistingDelete (CarlistingDeleteRequest) returns (google.protobuf.Empty);

  rpc CarlistingReadAll (google.protobuf.Empty) returns (CarlistingReadAllResponse);

  rpc CarlistingReadOne (CarlistingReadOneRequest) returns (CarListing);

  rpc CarlistingUpdate (CarlistingUpdateRequest) returns (CarListing);

}

message CarlistingCreateRequest {
  // Car listing to create
  CarListing carListing = 1;

}

message CarlistingDeleteRequest {
  // ID of the car listing to delete
  int64 listingId = 1;

}

message CarlistingReadAllResponse {
  repeated CarListing data = 1;
}

message CarlistingReadOneRequest {
  // ID of the car listing to retrieve
  int64 listingId = 1;

}

message CarlistingUpdateRequest {
  // ID of the car listing to update
  int64 listingId = 1;
  CarListing carListing = 2;

}

